Mage.Sets/src/mage/cards/a/AgainstAllOdds.java

```diff
@@ -0,0 +1,65 @@
+package mage.cards.a;
+
+import mage.abilities.Mode;
+import mage.abilities.effects.common.ExileTargetForSourceEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldTargetEffect;
+import mage.abilities.effects.common.ReturnToBattlefieldUnderOwnerControlTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.ComparisonType;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.ManaValuePredicate;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.common.TargetControlledPermanent;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author AhmadYProjects
+ */
+public final class AgainstAllOdds extends CardImpl {
+    private static final FilterControlledPermanent filter = new FilterControlledPermanent("creature card with mana value 3 or less");
+    private static final FilterCard filter2 = new FilterCard("artifact or creature card with mana value 3 or less from your graveyard");
+
+    static{
+        filter.add(Predicates.or(
+                CardType.CREATURE.getPredicate(),
+                CardType.ARTIFACT.getPredicate()
+        ));
+        filter.add(new ManaValuePredicate(ComparisonType.FEWER_THAN, 4));
+        filter2.add(new ManaValuePredicate(ComparisonType.FEWER_THAN,4));
+        filter2.add(Predicates.or(
+                CardType.CREATURE.getPredicate(),
+                CardType.ARTIFACT.getPredicate()
+        ));
+
+    }
+    public AgainstAllOdds(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{W}");
+        
+
+        // Choose one or both --
```

Comment:
don't forget to set the min and max modes

Reply:
fixed